### ISSUE TITLE
Crash in PythonQt::init() with Python 3.11.4, Qt 5.15, and PythonQt 3.4.2

### ISSUE THREAD

**Issue by richard42** [MeVisLab/pythonqt#107]
Hello,

I recently embarked on a quest to upgrade my project dependencies in a large media processing application called nextLab used internally for my employer, FotoKem Inc.  We were previously using PythonQt 3.2.0 with Qt 5.15.2 and Python 2.7 and it worked fine.  Many of our workstations are running MacOS, and newer releases of MacOS don't include Python 2.7 anymore, so I wanted to upgrade our PythonQt library to work with Python 3.

My initial test machine is running MacOS 10.15.7, and here I have installed Python 3.11.4 with Homebrew. I also have a workstation running MacOS 12, and the system Python interpreter on this one is also Python 3.11.4, so I'm hoping that I can support all versions of MacOS (10.15 through 12) with a single build of PythonQt, running against Python 3.11.

I downloaded the latest release of PythonQt (v3.4.2 from github), built the PythonQt library, and then built our application against this new PythonQt library.  Unfortunately however it crashed at runtime deep inside the PythonQt::init() function call. I have attached screenshots which show the call stack and source/disassembly listing at a couple of points along the way.

<img width="1537" alt="Screen Shot 2023-07-11 at 12 54 46 PM" src="https://github.com/MeVisLab/pythonqt/assets/6228255/eafd01c0-9b22-434f-9e4f-8a95a61978a9">
<img width="1537" alt="Screen Shot 2023-07-11 at 12 55 11 PM" src="https://github.com/MeVisLab/pythonqt/assets/6228255/c9a7ff48-52ec-42db-b60a-582dbf01608b">
<img width="1536" alt="Screen Shot 2023-07-11 at 12 55 46 PM" src="https://github.com/MeVisLab/pythonqt/assets/6228255/a67f1219-5b61-41ac-ad46-8a4439865677">

Just on a hunch, I commented out the code in PythonQt.cpp which imports the "asyncio" library if building against Python 3 (lines 105-111) and this fixed the crash.

**Comment by mrbean-bremen** [comment]
This is very strange - the code looks innocent enough and seems not to cause problems elsewhere.
You don't happen to have defined your own `asyncio` module or something like that? Can you reproduce this in a minimal setup where just the initialization is called?

**Comment by richard42** [comment]
Our application is huge and contains dozens of third party libraries so it is possible that something else is contributing to this problem. I'm sure that we are not using any custom "asyncio" python library though.  If I run the local python3 interpreter and just do an "import asyncio" it completes successfully with no errors. I will put together a simple test application and try to reproduce this in a more basic setting.

**Comment by richard42** [comment]
Ok I built a tiny test app to reproduce this problem and it was not reproducible with the test app.  However my system configuration also changed - during my earlier development I removed the "python@3.11" package from Homebrew and installed the standalone Python 3.11.4 package from the python.org installer. So maybe the Homebrew python install was contributing to this problem, or maybe it was something else in our full macos application. Either way, I'll close this issue report and open a new one in the future if I run into this problem again.
